src-plugins/medBinaryOperation/medBinaryOperatorBase.cpp

```
@@ -0,0 +1,251 @@
+        typedef itk::CastImageFilter< ImageType, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imA);
+        caster->Update();
+        imageA = caster->GetOutput();
+    }
+    else
+    {
+        imageA = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputA->data()));
+    }
+
+    if (m_inputB->identifier() != "itkDataImageUChar3")
+    {
+        typename ImageType2::Pointer imB = dynamic_cast< ImageType2*>((itk::Object*)(m_inputB->data()));
+        typedef itk::CastImageFilter< ImageType2, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imB);
+        caster->Update();
+        imageB = caster->GetOutput();
+    }
+    else
+    {
+        imageB = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputB->data()));
+    }
+    
+    if (imageA->GetLargestPossibleRegion().GetSize() != imageB->GetLargestPossibleRegion().GetSize())
+    {
+        return medAbstractProcess::DATA_SIZE;
+    }
+
+    typedef itk::InPlaceImageFilter< ImageTypeOutput, ImageTypeOutput >  FilterType;
+    typename FilterType::Pointer filter;
+    
+    if(description() == "AND")
+    {
+        typedef itk::AndImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> AndImageFilterType;
+        filter = AndImageFilterType::New();
+    }
+
+    else if(description() == "OR")
+    {
+        typedef itk::OrImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> OrImageFilterType;
+        filter = OrImageFilterType::New();
+    }
+
+    else if(description() == "XOR")
+    {
+        typedef itk::XorImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> XorImageFilterType;
+        filter = XorImageFilterType::New();
+    }
+
+    else
+    {
+        qDebug()<<__FILE__<<":"<<__LINE__<<", Wrong binary operator:"<<description();
```

This point of the program can only be reached through a programming error (i.e. implementing a binary operator without putting the support for it here, or making a mistake in the description function). It should throw an exception (with the same message).